### PR TITLE
feat: update dashboard filters tiles tab

### PR DESCRIPTION
**Pull request by joaoviana**
Closes half of #6072

left to do: ` Show the chart icon type next to the chart name `

### Description:

Refactor current implementation in `TileFilterConfiguration`;

Use `useListState` from `mantine hooks` to control group of checkboxes: example https://mantine.dev/core/checkbox/#indeterminate-state

Ensure there's always a defined field when checking a checkbox 

**in View mode**

https://github.com/lightdash/lightdash/assets/7611706/0e408219-49b7-4db9-b001-a12dddb9e68d

**in Edit mode**

https://github.com/lightdash/lightdash/assets/7611706/e8e4b438-22f0-4bbb-ba76-4697de5f6af0

### CODE REVIEW

File: packages/frontend/src/components/DashboardFilter/FilterConfiguration/TileFilterConfiguration.tsx [hunk 1]
```diff
@@ -27,123 +28,220 @@ interface TileFilterConfigurationProps {
...
+                checked: Boolean(isFilterAvailable && tileConfig),
+                tileUuid,
+                sortedFilters,
+                selectedFilter,
+            };
+        },
+    );
+
+    const [tileTargetCheckboxes, handlers] = useListState(
+        initialFilterTileTargets,
```
Comment (magnew) [inline packages/frontend/src/components/DashboardFilter/FilterConfiguration/TileFilterConfiguration.tsx:134]: Cool, I hadn't used this before.

**Review by magnew — APPROVED** [review]
Looks good to me. I played around with it, seems to work great.

File: packages/frontend/src/components/DashboardFilter/FilterConfiguration/TileFilterConfiguration.tsx (hunk 1 above)
Comment (joaoviana) [reply packages/frontend/src/components/DashboardFilter/FilterConfiguration/TileFilterConfiguration.tsx:134]: it's a really cool abstraction!

File: packages/frontend/src/components/DashboardFilter/FilterConfiguration/TileFilterConfiguration.tsx [hunk 2]
```diff
@@ -27,123 +28,220 @@ interface TileFilterConfigurationProps {
...
+                        event.currentTarget.checked
+                            ? FilterActions.ADD
+                            : FilterActions.REMOVE,
+                        value.tileUuid,
+                    );
+                }}
+            />
+
+            {value.sortedFilters && (
+                <Box ml={24} mt={6} display={!value.checked ? 'none' : 'auto'}>
```
[the remark this answers is not in the document]
Comment (joaoviana) [reply packages/frontend/src/components/DashboardFilter/FilterConfiguration/TileFilterConfiguration.tsx]: done!

**Review by PriPatel — APPROVED** [review]
Tested functionality - looks good! I'll open a separate ticket for minor styling tweaks.

**Pull request merged**